libs/sdk-common/src/input_parser.rs

```diff
@@ -197,6 +225,74 @@ pub async fn parse(
     Err(anyhow!("Unrecognized input type"))
 }
 
+fn validate_txt_record(decoded: &str) -> Option<bool> {
```

Comment:
This seems to always return either `None` or `Some(true)`. Probably can be replaced with `fn is_valid_txt_record(decoded: &str) -> bool`